src/librustc_typeck/check/wfcheck.rs

```diff
@@ -646,6 +687,18 @@ fn error_380(ccx: &CrateCtxt, span: Span) {
                Trait for ..`) must have no methods or associated items")
 }
 
+fn error_565(ccx: &CrateCtxt, span: Span) {
+    span_err!(ccx.tcx.sess, span, E0565,
+              "traits with default impls (`e.g. unsafe impl \
+               Trait for ..`) can not have predicates")
+}
+
+fn error_566(ccx: &CrateCtxt, span: Span) {
```

Comment:
(Though actually the better policy is to make a fresh tracking issue, I can do that.)